src/GraphQL/Types/ArgumentInformation.cs

```diff
@@ -69,30 +67,16 @@ protected ArgumentInformation(ParameterInfo parameterInfo, Type sourceType, Fiel
         /// </summary>
         public TypeInformation TypeInformation { get; }
 
-        private LambdaExpression? _expression;
         /// <summary>
-        /// Gets or sets a delegate in the form of a <see cref="LambdaExpression"/> to be used to populate
+        /// Gets or sets a delegate to be used to populate
         /// this method argument while building the field resolver.
         /// <br/><br/>
         /// If not set, a query argument will be added to the field and the argument's value will be used
         /// to populate the method argument while building the field resolver.
         /// <br/><br/>
-        /// The delegate must be of the type
-        /// <see cref="Expression{TDelegate}">Expression</see>&lt;<see cref="Func{T, TResult}">Func</see>&lt;<see cref="IResolveFieldContext"/>, TReturnType&gt;&gt;
-        /// where TReturnType matches <see cref="ParameterInfo">ParameterInfo</see>.<see cref="ParameterInfo.ParameterType">ParameterType</see>.
+        /// The delegate must be of the type Expression&lt;Func&lt;IResolveFieldContext,object&gt;&gt;.
         /// </summary>
-        public LambdaExpression? Expression
-        {
-            get => _expression;
-            set
-            {
-                if (value != null && (value.ReturnType != ParameterInfo.ParameterType || value.Parameters.Count != 1 || value.Parameters[0].Type != typeof(IResolveFieldContext)))
-                {
-                    throw new ArgumentException($"Value must be a lambda expression delegate of type Func<IResolveFieldContext, {ParameterInfo.ParameterType.Name}>.");
-                }
-                _expression = value;
-            }
-        }
+        public Expression<Func<IResolveFieldContext, object>>? Expression { get; set; }
```

Comment:
It may be changed back to `LambdaExpression` but it has no much sense. Note that there are no additional boxing introduced, only unboxing/convert from object to method parameter type - see MethodResolver.cs:38 
>  expressionBodies[i] = replaced.Type == methodParameters[i].ParameterType ? replaced : Expression.Convert(replaced, methodParameters[i].ParameterType);

It now happens when using `FromServicesAttribute` that sets `Expression` property. But since it returns some service from DI I think this service is ref type in 99,99% of cases.

Reply:
Only because FromServices happens to be a reference type.

Reply:
Yes.